demo/src/utils.js

```diff
@@ -12,3 +13,61 @@
 export function deepCopy(data) {
   return JSON.parse(JSON.stringify(data));
 }
+
+/**
+ * Computes the price of purchasing the given quantity
+ * of all items in the cart.
+ * @return {number} The total price.
+ */
+export function computePriceOfItemsInCart() {
+  const /** {ItemStore} */ items = store.getState().items;
+  let totalPrice = 0;
+  for (const item of Object.values(items)) {
+    totalPrice += item.quantity * item.cost;
+  }
+  return totalPrice;
+}
+
+/**
+ * Type definition for our application's standard item parameters.
+ * These are to be sent in the items array found in many gtag events.
+ * Acceptable parameters can be [found here](https://developers.google.com/gtagjs/reference/aw-events)
+ * @typedef {{item_id: string, item_name: string,
+  *     price: string}} ItemParameters
+  */
+
+/**
+  * Selects an item from the store via ID and returns an object containing
+  * its standard item parameters.
+  * @param {string} itemId
+  * @return {!ItemParameters}
+  */
+export function getItemParameters(itemId) {
+  const /** {ItemStore} */ items = store.getState().items;
+  const /** {number} */ quantity = items[itemId].quantity;
+  return {
+    item_id: itemId,
+    item_name: items[itemId].name,
+    price: items[itemId].cost,
+    // if quantity is positive, include it in the item parameter
+    ...(quantity > 0 && {quantity}),
```

Comment:
Nifty!

Reply:
Thanks! ES6 has some fun stuff. Of course it isn't as readable hence the comment but it gets the job done :)

Reply:
Yep, as long as you comment, I'm good with this ES6 "magic".